strings/base_collections_vector.h

```diff
@@ -115,13 +115,27 @@ namespace winrt::impl
 
         bool IndexOf(Windows::Foundation::IInspectable const& value, uint32_t& index) const
         {
-            try
+            if constexpr (is_com_interface_v<T>)
             {
-                return IndexOf(unbox_value<T>(value), index);
+                if (!value)
+                {
+                    return base_type::IndexOf(nullptr, index);
+                }
+                else if (auto as = value.try_as<T>())
```

Comment:
I would prefer that the value-type branch be indented at the same level as the COM case, so that it's clear that we are dealing with two cases, not three.

```cpp
if constexpr (is_com_interface_v<T>)
{
    if (!value)
    {
        return base_type::IndexOf(nullptr, index);
    }
    else if (auto as = value.try_as<T>())
    {
        return base_type::IndexOf(as, index);
    }
    else
    {
        return false;
    }
}
else
{
    if (auto as = value.try_as<T>())
    {
        return base_type::IndexOf(as.value(), index);
    }
    else
    {
        return false;
    }
}
```

Could factor out the final "return false".

```cpp
if constexpr (is_com_interface_v<T>)
{
    if (!value)
    {
        return base_type::IndexOf(nullptr, index);
    }
    else if (auto as = value.try_as<T>())
    {
        return base_type::IndexOf(as, index);
    }
}
else
{
    if (auto as = value.try_as<T>())
    {
        return base_type::IndexOf(as.value(), index);
    }
}
return false;
```

Reply:
good idea - done